ledger/src/blockstore.rs

```diff
@@ -1490,10 +1628,11 @@ impl Blockstore {
 
             if !self.has_duplicate_shreds_in_slot(slot) {
                 let ending_shred: Vec<u8> = self
-                    .get_data_shred_from_just_inserted_or_db(
+                    .get_shred_from_just_inserted_or_db(
                         just_inserted_shreds,
                         slot,
-                        last_index.unwrap(),
+                        u32::try_from(last_index.unwrap()).unwrap(),
```

Comment:
this is annoying, it seems in order to use `u32` indices everywhere, we need to update the `slot_meta` to use `u32` for `last_index, consumed, received`. then we can also update the shred columns to use `u32`